docs: add standing convention for naming forks of projects (#17374)
conan-io/conan-center-index: docs/faqs.md

For example, `GSL` is the name of `Guidelines Support Library` from Microsoft and `GNU Scientific Library` from GNU. Both libraries are commonly known as `gsl`, however, to disambiguate (if there is already a `gsl` package in this repo) we could use `ms-gsl` in the first case or `gnu-gsl` in the second.

## What is the policy for naming forks?

> When submitting recipes, its important to pick names which clearly identifies the source. For projects which are forks of existing but diverge significantly enough

Reviewer: ```suggestion
When submitting recipes, it's important to pick names which clearly identify the source. For projects which are forks of existing but diverge significantly enough
```

> to be unique (subject to opinion), the should be prefixed with the author or organization such as `author-name`. A good example is the [crow](https://github.com/ipkn/crow)

Reviewer: ```suggestion
to be unique (subject to opinion), they should be prefixed with the author or organization such as `author-name`. A good example is the [crow](https://github.com/ipkn/crow)
```

which was forked [crowcpp-crow](https://github.com/CrowCpp/Crow). The original project maintain the `crow` name where as the more featured and advanced fork is `crowcpp-crow`.

> Forks are acceptable submissions given they have significantly divergent fro mthe upstream. This can include adding new features, updateing compiler or standard support, and actively being maintained. Changes which are not sufficient are build scripts, small bug fixes, "conan-ization" - all of these can be

Reviewer: ```suggestion
Forks are acceptable submissions given they have significantly divergent from the upstream. This can include adding new features, updating compiler or standard support, and actively being maintained. Changes which are not sufficient are build scripts, small bug fixes, "conan-ization" - all of these can be
```
